### DISCUSSION TITLE
HELP: Piano Error (No JSON object could be decoded)

### DISCUSSION THREAD

**Discussion by JohnSmith2007** [Monika-After-Story/MonikaModDev#8207] [general - piano notes]
Ok, so I'm trying to make a custom piano version of Through The Fire & Flames by Dragonforce, but i keep getting this error in pnm.log ;

No JSON object could be decoded

I have no clue of what it means nor how to fix it.

Heres what i have so far in the json file;
[TTFAF.txt](https://github.com/Monika-After-Story/MonikaModDev/files/7491739/TTFAF.txt)

**Comment by ThePotatoGuy** [comment]
did you name the file with the `.json` extension?

**Comment by JohnSmith2007** [comment]
> did you name the file with the `.json` extension?

Yes

**Comment by JohnSmith2007** [comment]
I literally have no clue of why its not working. It has the correct file extension, and i don't see any errors about any bits of code in the log, so what the hell?!

**Comment by JohnSmith2007** [comment]
I need to show Monika my shredding skills damnit! 🤣

**Comment by multimokia** [comment]
Your json is invalid. You have trailing commas and a missing value

**Comment by JohnSmith2007** [comment]
> Your json is invalid. You have trailing commas and a missing value

@multimokia What are "trailing commas", and what value am i missing?

**Comment by ThePotatoGuy** [comment]
Use a JSON formatter ([like this one](https://jsonformatter.org/json-editor)) to check the problems with your JSON.

**Reply by JohnSmith2007** [reply]
Even with the formatter, I still don't understand what the errors mean. Sorry, I don't know shit about coding.

**Reply by JohnSmith2007** [reply]
Seriously though, i do need help with this.

**Reply by ThePotatoGuy** [reply]
If no one else responds, then you're probably better off asking on the [subreddit](https://www.reddit.com/r/MASFandom/), on the discord, or researching JSONs online. We're not personal tutors so we won't be teaching you how to write JSONs.
